community/cypher/cypher-planner-3.5/src/main/java/org/neo4j/cypher/internal/compiler/v3_5/common/CypherOrderability.java

```diff
@@ -121,7 +121,7 @@ else if ( lhs instanceof AnyValue )
         }
         else if ( rhs instanceof AnyValue )
         {
-            AnyValue lhsValue = (lhs instanceof AnyValue) ? (AnyValue) lhs : ValueUtils.of( lhs );
+            AnyValue lhsValue = ValueUtils.of( lhs );
```

Comment:
This would be a minor performance regression, otherwise these changes LGTM :)

Reply:
Also, for us to merge this you have to sign the [CLA](https://neo4j.com/developer/cla/)

Reply:
Thanks. Will do.

Reply:
If I look at this: https://lgtm.com/projects/g/neo4j/neo4j/snapshot/5817c4b32a4788673a642833d05cd1eb7fe8a573/files/community/cypher/cypher-planner-3.5/src/main/java/org/neo4j/cypher/internal/compiler/v3_5/common/CypherOrderability.java#xaef4970408ffb808:1
I don't think it would be a performance regression...

Reply:
Oh. Right. Never mind.